### PR TITLE
Issue 18348 d

### PR DESCRIPTION
**Pull request by jdotcms**
- Adding fixes for drag and drop
- Adding fixes for content editor with dotAsset

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/util/UtilHTML.java [hunk 1]
```diff
@@ -442,9 +442,10 @@ public static String getIconClass(final Contentlet contentlet) throws DotDataExc
...
-                return "uknIcon " + UtilMethods.getFileExtension(
-                		Try.of(()-> contentlet.getBinary("asset").getName())
-								.getOrElse("ukn") )+ "Icon";
+            	final String dotUri  = Try.of(()-> contentlet.getBinary("asset").getName()).getOrElse("ukn");
+            	final String dotIcon = UtilMethods.getFileExtension(dotUri);
+
+				return dotUri.equals(dotIcon)? "uknIcon": dotIcon + "Icon";
             default:
                 return IconType.UNKNOWN.iconName();
         }
```
Comment (freddyucv) [inline dotCMS/src/main/java/com/dotmarketing/util/UtilHTML.java:451]: any test?

**Review by freddyucv — CHANGES_REQUESTED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java [hunk 2]
```diff
@@ -648,6 +653,23 @@ else if(cont.getStructure().getStructureType()==Structure.STRUCTURE_TYPE_HTMLPAG
...
+
+					final Map<String, Object> fileMap = cont.getMap();
+					fileMap.put("mimeType", mimeType);
+					fileMap.put("path",          "/dA/" + cont.getIdentifier() + StringPool.SLASH);
+					fileMap.put("type", "dotasset");
+					fileMap.put("type", "dotasset");
+					fileMap.put("name",     fileName);
+					fileMap.put("fileName", fileName);
+
+					return fileMap;
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java:671]: Don't we have a transformer you can use for this @jdotcms ?

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/util/UtilHTMLTest.java [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        assertEquals("vanityIcon", iconClass);
+    }
+
+    @Test
+    public void test_getIconClass_dotasset_txt () throws DotDataException, IOException {
+
+        final Contentlet contentlet = mock(Contentlet.class);
+        final ContentType type      = mock(ContentType.class);
+        final BaseContentType any   = BaseContentType.DOTASSET;
+        File txtFile                = FileUtil.createTemporalFile("testtxt", ".txt");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/util/UtilHTMLTest.java:165]: Codacy Issue found: [Local variable 'txtFile' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5436906)

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/util/UtilHTMLTest.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+     * Given Scenario: Given a dotasset base type and txt file
+     * ExpectedResult: txtIcon icon should be returned
+     */
+    @Test
+    public void test_getIconClass_dotasset_txt () throws DotDataException, IOException {
+
+        final Contentlet contentlet = mock(Contentlet.class);
+        final ContentType type      = mock(ContentType.class);
+        final BaseContentType any   = BaseContentType.DOTASSET;
+        File txtFile                = FileUtil.createTemporalFile("testtxt", ".txt");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotmarketing/util/UtilHTMLTest.java:223]: Codacy Issue found: [Local variable 'txtFile' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5436906)

File: dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/portlets/browser/ajax/BrowserAjax.java:671]: if not exists any, Should we create a new transformer?

**Review by freddyucv — APPROVED** [review]

**Pull request merged**